enhancements/api-review/custom-tags-aws.md

```diff
@@ -79,65 +99,323 @@ The userTags field is intended to be set at install time and is considered immut
+
+For installer configuration compatibility, configuration with `experimentalPropagateUserTags` should be supported.
 
 On downgrade:
 
-The status field may remain populated, components may or may not continue to tag newly created resources w/ the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
+The status/spec field may remain populated, components may or may not continue to tag newly created resources with the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
 
 ### Version Skew Strategy
 
-TBD
-
 ## Implementation History
 
 ## Drawbacks
 
-If a customer decides they do not want these tags applied to new resources, there is no clean way to address that need:
+If a user decides that they do not want these tags applied to new resources, there is no clean way to address that need:
 
-1. they would have to get help from support to edit the status field
-2. they would have to manually remove the undesired tags from any existing resources
-
-In the future we will want to introduce a spec field where a customer can specify and edit these tags, which will be reflected into status when changed, and we will expect consuming components to reconcile changes to the set of tags by applying net new tags to existing resources (but they will still not remove tags that are dropped from the list of tags).
+1. User would have to get help from support to edit the status field for older version which uses `experimentalPropagateUserTags`.
+2. User would have to manually remove the undesired user-defined tags from some existing resources even though user-defined tags can be marked for deletion by setting tag value to empty string.
+3. User would have to manually update user-defined tags for AWS resources which are not managed by an operator.
+4. User would have to manually update user-defined tags for AWS resources where update logic is not supported by an operator.
 
 ## Alternatives
 
-TBD
+An operator can be designed to behave similar to how the installer's destroy behaves, namely by using the resource tagging API to search for resources with the kubernetes.io/.../cluster=owned tag.
+The operator would reconcile on changes to the infrastructure CR to ensure that resources owned by the cluster have tags matching what is in the infrastructure CR.
+That would get tricky when it comes to resources owned by in-cluster operators, though, particularly with tags that may have different values in the corresponding kube resources.
+The operator would need to monitor multiple kube resources, infrastructure CR and multiple instances of different AWS resource types.
 
 ## Infrastructure Needed [optional]
 
-TBD
+## Future Work
+
+1. Update the Infrastructture CRD to enable validation  to check if any tag that starts with `kubernetes.io` or `openshift.io using
```

Comment:
You've missed the closing backtick after openshift.io

Reply:
Thanks for locating, updated the tick.